### PR TITLE
Fix index error if datafields outnumber values

### PR DESCRIPTION
**Pull request by makermelissa**
Fixes https://github.com/adafruit/Adafruit_Learning_System_Guides/issues/1553.

### CODE REVIEW

File: adafruit_portalbase/__init__.py [hunk 1]
```diff
@@ -392,7 +392,7 @@ def _fill_text_labels(self, values):
         if self._text:
             value_index = 0  # In case values and text is not the same
             for i in range(len(self._text)):
-                if not self._text[i]["is_data"]:
+                if not self._text[i]["is_data"] or value_index > len(values) - 1:
```
Comment (ladyada) [inline adafruit_portalbase/__init__.py:395]: wanna add some parens here?

**Review by ladyada — CHANGES_REQUESTED** [review]

File: adafruit_portalbase/__init__.py [hunk 2]
```diff
@@ -392,7 +392,7 @@ def _fill_text_labels(self, values):
         if self._text:
             value_index = 0  # In case values and text is not the same
             for i in range(len(self._text)):
-                if not self._text[i]["is_data"]:
+                if not self._text[i]["is_data"] or (value_index > len(values) - 1):
```
Comment (ladyada) [inline adafruit_portalbase/__init__.py:395]: if (not self._text[i]["is_data"]) or (value_index > (len(values) - 1)):

Comment (makermelissa) [reply adafruit_portalbase/__init__.py:395]: gotcha, thanks

File: adafruit_portalbase/__init__.py [hunk 3]
```diff
@@ -392,7 +392,7 @@ def _fill_text_labels(self, values):
         if self._text:
             value_index = 0  # In case values and text is not the same
             for i in range(len(self._text)):
-                if not self._text[i]["is_data"] or (value_index > len(values) - 1):
+                if (not self._text[i]["is_data"]) or (value_index > len(values) - 1):
```
Comment (ladyada) [inline adafruit_portalbase/__init__.py:395]: one more!
if (not self._text[i]["is_data"]) or (value_index > (len(values) - 1)):

**Review by ladyada — CHANGES_REQUESTED** [review]

File: adafruit_portalbase/__init__.py (hunk 3 above)
Comment (makermelissa) [reply adafruit_portalbase/__init__.py]: Oops, thanks

**Review by ladyada — APPROVED** [review]

**Pull request merged**